examples/policy_gradient/reinforce/utils.py

```diff
@@ -32,17 +33,51 @@ def concatenate(weights_list):
 def shuffle(trajectory):
   permutation = np.random.permutation(trajectory["dones"].shape[0])
   for key, val in trajectory.items():
-    trajectory[key] = val[permutation][permutation]
+    trajectory[key] = val[permutation]
   return trajectory
 
 
-def iterate(trajectory, batchsize):
-  trajectory = shuffle(trajectory)
-  curr_index = 0
-  # TODO(pcm): This drops some data at the end of the batch.
-  while curr_index + batchsize < trajectory["dones"].shape[0]:
-    batch = dict()
-    for key in trajectory:
-      batch[key] = trajectory[key][curr_index:(curr_index + batchsize)]
-    curr_index += batchsize
-    yield batch
+def make_divisible_by(array, n):
+  return array[0:array.shape[0] - array.shape[0] % n]
+
+
+def average_gradients(tower_grads):
+  """Calculate the average gradient for each shared variable across all towers.
+  Note that this function provides a synchronization point across all towers.
```

Comment:
- the first line of documentation should fit in one line
- add a newline before Args
- add a newline before Returns